wled00/FX_fcn.cpp

```diff
@@ -837,13 +866,101 @@ uint32_t Segment::getPixelColor(int i)
   }
 #endif
 
+  return strip.getPixelColor(getPixelIndex(i));
+}
+
+uint32_t Segment::getBufferColor(int i) {
+  return strip._buffer[getPixelIndex(i)];
+}
+
+uint16_t Segment::getPixelIndex(int i) {
+  i &= 0xFFFF;
   if (reverse) i = virtualLength() - i - 1;
   i *= groupLength();
   i += start;
   /* offset/phase */
   i += offset;
   if ((i >= stop) && (stop>0)) i -= length(); // avoids negative pixel index (stop = 0 is a possible value)
-  return strip.getPixelColor(i);
+  return i;
+}
+
+void Segment::blendTransition() {
+  switch (transitionStyle) {
+    case TRANSITION_STYLE_PUSH_RIGHT: {
+      uint16_t len = virtualLength();
+      uint16_t pos = (float(progress()) / float(0xFFFFU)) * len;
```

Comment:
Avoid using `float` (or `double`).

Reply:
Yeah that was still on my todo list. Thanks for reminding me! :)